libs/net/dal/Services/UserService.cs

```diff
@@ -40,6 +40,11 @@ public IPaged<User> Find(UserFilter filter)
             query = query.Where(c => EF.Functions.Like(c.FirstName.ToLower(), $"{filter.FirstName.ToLower()}%"));
         if (!String.IsNullOrWhiteSpace(filter.LastName))
             query = query.Where(c => EF.Functions.Like(c.LastName.ToLower(), $"{filter.LastName.ToLower()}%"));
+        if (!String.IsNullOrWhiteSpace(filter.Keyword))
+            query = query.Where(c => EF.Functions.Like(c.Username.ToLower(), $"{filter.Keyword.ToLower()}%") || EF.Functions.Like(c.Email.ToLower(), $"{filter.Keyword.ToLower()}%") || EF.Functions.Like(c.FirstName.ToLower(),
+            $"{filter.Keyword.ToLower()}%") || EF.Functions.Like(c.LastName.ToLower(), $"{filter.Keyword.ToLower()}%"));
+        if (!String.IsNullOrWhiteSpace(filter.RoleName))
+            query = query.Where(c => c.Roles.Any(r => r.Name.ToLower() == filter.RoleName.ToLower()));
```

Comment:
This doesn't seem to be working, is there something obvious I am doing wrong here? @Fosol

Reply:
Search by role isn't working?  You'd have to debug the API and see the SQL it's generating.

Reply:
Fixed it :)